### PR TITLE
Remove gratuitous JSON.stringify from parseHandles utility

### PR DESCRIPTION
**Pull request by markfields**
## Description

We added the `encode` and `decode` functions onto `IFluidSerializer` a few years back, and updated the utility `makeHandlesSerializable` to use `encode`, but left the inverse utility `parseHandles` using the old (less efficient) way of doing `serializer.parse(JSON.stringify(value))`.

This PR switches to using `serializer.encode` which does an object walk itself, and avoids translating through `string`.

## Reviewer Guidance

I'm assuming there is adequate test coverage of this code path.  I also haven't run any benchmarks - @DLehenbauer any risk that somehow this is _slower_ than the roundtrip through `string`?

### Merge-Tree fuzz test failures

Merge-Tree fuzz tests seeds 29 and 47 are failing with assert `0x04f - "minSeq of collab window > target minSeq!"`.  This is surprising but they both failed for two runs in a row, so something is going on here.  I'm not able to look into it for now so converting to a draft PR.

### CODE REVIEW

**Review by DLehenbauer — APPROVED** [review]

File: packages/framework/attributor/src/test/attribution/documents/default/attributor-lz4-and-delta-snap.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
...
...,\"props\":{\"prop2\":\"dHWn1\",\"prop3\":\"5KoWp\"},\"type\":2},{\"pos1\":31,\"pos2\":40,\"props\":{\"prop2\":\"dHWn1\",\"prop3\":\"5KoWp\"},\"type\":2}],\"type\":3},\"referenceSequenceNumber\":81,\"type\":\"op\",\"sequenceNumber\":82,\"minimumSequenceNumber\":76},{\"clientId\":\"d3fdea22-7906-496f-9fc5-0bf988954452\",\"clientSequenceNumber\":29,\"contents\":{\"pos1\":31,\"seg\":\"yc5Bz\",\"type\":0},\"referenceSequenceNumber\":82,\"type\":\"op\",\"sequenceNumber\":83,\"minimumSequenceNumber\":76},{\"clientId\":\"d3fdea22-7906-496f-9fc5-0bf988954452\",\"clientSequenceNumber\":30,\"contents\":{\"pos1\":36,\"seg\":\"mQny\",\"type\":0},\"referenceSequenceNumber\":83,\"type\":\"op\",\"sequenceNumber\":84,\"minimumSequenceNumber\":76},{\"clientId\":\"25adaa92-9220-49e2-a852-3c7e97ac9d1c\",\"clientSequenceNumber\":24,\"contents\":{\"ops\":[],\"type\":3},\"referenceSequenceNumber\":84,\"type\":\"op\",\"sequenceNumber\":85,\"minimumSequenceNumber\":76},{\"clientId\":\"a2df62bb-461f-4c2e-a3ad-fc5df463c464\",\"clientSequenceNumber\":29,\"contents\":{\"pos1\":4,\"seg\":\"x2NAdmR\",\"type\":0},\"referenceSequenceNumber\":85,\"type\":\"op\",\"sequenceNumber\":86,\"minimumSequenceNumber\":76}]"
```
Comment (markfields) [inline packages/framework/attributor/src/test/attribution/documents/default/attributor-lz4-and-delta-snap.json:13]: Per @Abe27342 this change to what's output here is benign.

https://github.com/microsoft/FluidFramework/pull/19242/files#r1479094891

**Comment by markfields** [comment]
@connorskees - Funny - this doesn't fail those fuzz tests anymore...

@DLehenbauer, @Abe27342 - Any thoughts on whether this is a safe change to make?

**Comment by Abe27342** [comment]
> @connorskees - Funny - this doesn't fail those fuzz tests anymore...
> 
> @DLehenbauer, @Abe27342 - Any thoughts on whether this is a safe change to make?

I'd want to understand the source of failure first. Do you remember exactly which tests were failing? i.e. were they the ones in sequence? or some of the test farms in merge-tree?

We've recently changed the scenarios fuzz tests exercise to support the stashed op flow, so ones built on top of the DDS fuzz harness won't be exercising the same scenarios they did 1 month ago, but the issue could still be prevalent (especially if it was only 2 seeds)

**Comment by connorskees** [comment]
https://github.com/microsoft/FluidFramework/pull/19242 may have had an effect in making those tests no longer fail

**Comment by markfields** [comment]
@Abe27342 can you link the commit(s) that changed the fuzz harness?  I'd like to try the combo of Connor's PR + old fuzz harness and see which tweaked it.

**Comment by markfields** [comment]
@vladsud - This is one gratuitous string roundtrip that I was talking about this morning.

**Comment by Abe27342** [comment]
@markfields #20508 and various follow-ups to use `random.handle()` in different packages (these are made by Jillian, check on github), #20358, #19802, #19701, etc.... It'll be hard to narrow down

**Comment by vladsud** [comment]
Is there some write up that tells us where we were, where we are today, and what else we can do to make things better?
My main concern - when I did profile matrix 2 months back (at large scale), almost all the time in profiler was handle parsing.
Should it be better now? I'll re-meassure, but having some English narrative somewhere with status would be very useful (including assessment of possible next steps and SWAGs)

**Comment by markfields** [comment]
> Is there some write up that tells us where we were, where we are today, and what else we can do to make things better? My main concern - when I did profile matrix 2 months back (at large scale), almost all the time in profiler was handle parsing. Should it be better now? I'll re-meassure, but having some English narrative somewhere with status would be very useful (including assessment of possible next steps and SWAGs)

I'll write down e2e op submit/processing perf benchmarking / optimization for next semester planning.

**Comment by markfields** [comment]
Closing this in favor of #21102. At this time, not planning to dig into the fuzz test failures.  But open to feedback that we should prioritize that investigation.  It's hard for me to imagine where we would have a correctness dependency on the LHS here - we've been using the `serializer.encode` in production for a long time which does the same object walk.  And @jzaffiro added `serialier.decode` calls to the DDS fuzz tests last month, and we aren't seeing problems since then.

**Pull request closed without merging**